cpmpy/transformations/flatten_model.py

```diff
@@ -185,6 +185,34 @@ def flatten_constraint(expr):
     - Numeric inequality (>=,>,<,<=,): Numexpr >=< Var     (CPMpy class 'Comparison')
     - Reification (double implication): Boolexpr == Var    (CPMpy class 'Comparison')
         """
+        left, right = expr.args[0], expr.args[1]
+
+        if isinstance(left, Operator) and left.name == "wsum" and any(_should_wsum(xi) for xi in left.args[1]):
+            w, x = left.args[0], left.args[1]
+            w_new, x_new = [], []
+            for wi, xi in zip(w, x):
+                if _should_wsum(xi):
+                    wni, xni = _wsum_make(xi)
+                    wni = [wnij * wi for wnij in wni]
+                    w_new += wni
+                    x_new += xni
+                else:
+                    w_new.append(wi)
+                    x_new.append(xi)
+
+            return flatten_constraint(
```

Comment:
its a bit unusual to want to flatten something that you just created. Can't you ensure the arguments are flattened, and then just create your expression?